lighthouse-core/test/runner-test.js

```diff
@@ -6,16 +6,83 @@
 'use strict';
 
 const Runner = require('../runner');
+const GatherRunner = require('../gather/gather-runner');
 const driverMock = require('./gather/fake-driver');
 const Config = require('../config/config');
 const Audit = require('../audits/audit');
+const assetSaver = require('../lib/asset-saver');
 const assert = require('assert');
 const path = require('path');
+const sinon = require('sinon');
+
 const computedArtifacts = Runner.instantiateComputedArtifacts();
 
 /* eslint-env mocha */
 
 describe('Runner', () => {
+  const saveArtifactsSpy = sinon.spy(assetSaver, 'saveArtifacts');
+  const loadArtifactsSpy = sinon.spy(assetSaver, 'loadArtifacts');
```

Comment:
oh these are straight spies, I thought you were stubbing

I'd have a slight preference for stubbing `loadArtifacts` and asserting the LHR looks like we'd expect it to than 100% relying on 'was called' assertions

Reply:
i think we'll have to discuss this in person, as all runner tests do actually run gatherrunner.run, runAudits, generateReport, etc. So only going with stubs for these seems inconsistent. 
Also, we'll have to mimic what these methods do on the test side if we want to see that everything works (for example: does -GA save to disk?)

Reply:
I'm all for having at least some integration testing of this 👍 

Maybe we'll chat in person but seems like there's room for both